tests/ui/deriving/deriving-all-codegen.rs

```diff
@@ -45,6 +45,23 @@ struct Big {
     b1: u32, b2: u32, b3: u32, b4: u32, b5: u32, b6: u32, b7: u32, b8: u32,
 }
 
+// `PartialEq` must reorder scalar types to be first and preserve the order of
+// non-scalar types. In contrast, `PartialOrd` must compare by order of declaration.
```

Comment:
The first sentence is misleading. It would be better to say something like "It is more efficient to compare scalar types before non-scalar types." Likewise for the similar comments below.